.github/workflows/version-preview.yaml

```diff
@@ -0,0 +1,118 @@
+
+defaults:
+  run:
+    shell: bash
+
+permissions:
+  contents: read
+  pull-requests: write
+
+jobs:
+  preview:
+    name: Preview Release
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v4
+        with:
+          fetch-depth: 0
+
+      - name: Fetch base branch
+        run: git fetch origin ${{ github.base_ref }}
+
+      - name: Analyze PR Commits
+        id: analyze
+        run: |
+          # Get commits from this PR
+          COMMITS=$(git log origin/${{ github.base_ref }}..HEAD --pretty=format:"%s")
+          echo "Commits in this PR:"
+          echo "$COMMITS"
+
+          # Determine release type
+          if echo "$COMMITS" | grep -qE "^[a-z]+(\(.+\))?!:|BREAKING CHANGE:"; then
+            TYPE="major"
+          elif echo "$COMMITS" | grep -qE "^feat(\(.+\))?:"; then
+            TYPE="minor"
+          elif echo "$COMMITS" | grep -qE "^fix(\(.+\))?:"; then
+            TYPE="patch"
+          else
+            TYPE="none"
+          fi
+
+          echo "release_type=$TYPE" >> $GITHUB_OUTPUT
+          echo "Release type: $TYPE"
+
+          # Calculate next version if releasing
+          if [ "$TYPE" != "none" ]; then
+            CURRENT=$(git tag --sort=-v:refname | head -1 | sed 's/^v//')
+            IFS='.' read -r maj min pat <<< "$CURRENT"
+
+            [ "$TYPE" = "major" ] && maj=$((maj + 1)) && min=0 && pat=0
+            [ "$TYPE" = "minor" ] && min=$((min + 1)) && pat=0
+            [ "$TYPE" = "patch" ] && pat=$((pat + 1))
+
+            VERSION="$maj.$min.$pat"
+            echo "next_version=$VERSION" >> $GITHUB_OUTPUT
+            echo "will_release=true" >> $GITHUB_OUTPUT
+            echo "Next version: v$VERSION"
+
+            # Generate notes
+            {
+              echo "release_notes<<EOF"
+              echo "## Changes in this PR"
+              echo ""
+              echo "$COMMITS" | sed 's/^/- /'
+              echo "EOF"
+            } >> $GITHUB_OUTPUT
+          else
+            echo "will_release=false" >> $GITHUB_OUTPUT
+          fi
+
+      - name: Comment PR
+        uses: actions/github-script@v7
+        with:
+          script: |
+            const willRelease = '${{ steps.analyze.outputs.will_release }}';
+            let body = '';
+
+            if (willRelease === 'true') {
+              const releaseType = '${{ steps.analyze.outputs.release_type }}';
+              const version = '${{ steps.analyze.outputs.next_version }}';
+              const notes = `${{ steps.analyze.outputs.release_notes }}`;
+
+              body = `## Release Preview\n\n**Release Type:** \`${releaseType}\`\n**Next Version:** \`v${version}\`\n\n### Release Notes\n${notes}\n\n---\n*This shows what release will be created when this PR is merged.*`;
+            } else {
+              body = '## Release Preview\n\nNo new release will be created from this PR.\n\nThis PR does not contain conventional commits that trigger a release.';
```

Comment:
would suggest using existing action instead of writing out own script. Please refer to how terraform output is commented on the PR.